service/history/shard/context_test.go

```diff
@@ -108,10 +108,9 @@ func (s *contextSuite) SetupTest() {
 }
 
 func (s *contextSuite) TestOverwriteScheduledTaskTimestamp() {
-	now := s.timeSource.Now()
+	now := time.Now()
```

Comment:
Why use that instead of the faked time source?

Reply:
The original implementation
```
now := s.timeSource.Now()
s.timeSource.Update(now)
```
doesn't really make sense. It gets the time from the fake time source and uses it to update the fake time source, so a no-op.

We don't have to update the fake time source in this test, then every run of the test will use the exact same timestamp. But I'd like to introduce some randomness into the test to increase the coverage.